packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -215,7 +215,8 @@ void YogaLayoutableShadowNode::adoptYogaChild(size_t index) {
   } else {
     // The child is owned by some other node, we need to clone that.
     // TODO: At this point, React has wrong reference to the node. (T138668036)
-    auto clonedChildNode = childNode.clone({});
+    auto clonedChildNode = std::static_pointer_cast<YogaLayoutableShadowNode>(childNode.clone({}));
+    clonedChildNode->invalidateLayoutCaches();
```

Comment:
I *think* this can be moved to the constructors of relevant nodes, to not invalidate every node, though I haven't explored that yet. WDYT?

Reply:
for sure, if we can avoid to invalidate all the children and only invalidate relevant nodes, that would be better. It will let us keep some perf improvements!

Reply:
also, shouldn't we invalidate the nodes only when the FontMultiplier changes?

Reply:
> also, shouldn't we invalidate the nodes only when the FontMultiplier changes?

We should but I don't think we have access to any other node that we could compare with here.